### PR TITLE
Sync: Engine API and remaining triggers

### PR DESCRIPTION
**Pull request by malmstein**
Task/Issue URL: https://app.asana.com/0/0/1204239356024394/f

### Description
This PR cleans up several pieces from the Sync Engine in consequence of the updated Tech Design.
Added background sync
Added sync triggered by opening Bookmarks screen

### Steps to test this PR
Details in task

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204255299433079

### CODE REVIEW

**Comment by malmstein** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3039** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3039" target="_blank">Graphite</a> 
    * **PR #3044** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3044" target="_blank">Graphite</a> 
      * **PR #3106** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3106" target="_blank">Graphite</a> 
        * **PR #3180** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3180" target="_blank">Graphite</a>  👈
        * **PR #3147** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3147" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3180?utm_source=stack-comment).

File: sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/engine/SyncScheduler.kt [hunk 1]
```diff
@@ -43,10 +44,12 @@ class RealSyncScheduler @Inject constructor(private val syncStateRepository: Syn
                 IN_PROGRESS -> DISCARD
                 FAIL -> EXECUTE
                 else -> {
-                    val syncTime = Instant.parse(lastSync.timestamp)
-                    val now = Instant.now()
-                    val duration = Duration.between(syncTime, now)
-                    if (duration.toMinutes() > DEBOUNCE_PERIOD_IN_MINUTES) {
+                    val syncTime = OffsetDateTime.parse(lastSync.timestamp)
```
Comment (cmonfortep) [inline sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/engine/SyncScheduler.kt:47]: Based on a conversation yesterday. This made sense when we were doing one single operation for all features, now that we do sync per feature, I think this should be revisited. For example: FAIL refers to all operations or single feature, one feature failing will mark syncstate as fail or success? will it depend on the response code? etc.

Can you create a class somewhere to don't miss this recent change? I'm ok not working on this now, but we should have it in the roadmap

File: versions.properties [hunk 2]
```diff
@@ -134,3 +136,5 @@ version.com.journeyapps..zxing-android-embedded=4.3.0
 version.com.google.zxing..core=3.3.0
 
 version.androidx.room.testing=2.4.3
+
+version.org.json.json=20230227
```
Comment (cmonfortep) [inline versions.properties:140]: This seems duplicated from the one at the top

File: saved-sites/saved-sites-store/src/main/java/com/duckduckgo/savedsites/store/SavedSitesEntitiesDao.kt [hunk 3]
```diff
@@ -97,7 +97,7 @@ interface SavedSitesEntitiesDao {
         lastModified: String = DatabaseDateFormatter.iso8601(),
     )
 
-    @Query("select * from entities order by rowid desc limit 1")
+    @Query("select * from entities limit 1")
```
Comment (cmonfortep) [inline saved-sites/saved-sites-store/src/main/java/com/duckduckgo/savedsites/store/SavedSitesEntitiesDao.kt:100]: unsure, so just highlighting this, if we expect "lastmodified" not sure why we remove `order by`

**Review by cmonfortep — APPROVED** [review]
LGTM, we discussed changes offline yesterday so sync engine supports syncing per feature. That will be done in another PR, so approving this one.

**Review by cmonfortep — APPROVED** [review]
LGTM, we discussed changes offline yesterday so sync engine supports syncing per feature. That will be done in another PR, so approving this one.

**Review by cmonfortep — APPROVED** [review]
LGTM, we discussed changes offline yesterday so sync engine supports syncing per feature. That will be done in another PR, so approving this one.

**Pull request merged**